languagetool-core/src/main/java/org/languagetool/tagging/disambiguation/rules/DisambiguationRuleLoader.java

```diff
@@ -44,6 +44,9 @@ public final List<DisambiguationPatternRule> getRules(InputStream stream, Langua
     DisambiguationRuleHandler handler = new DisambiguationRuleHandler(language, xmlPath);
     SAXParserFactory factory = SAXParserFactory.newInstance();
     SAXParser saxParser = factory.newSAXParser();
+    saxParser.getXMLReader().setProperty("jdk.xml.maxGeneralEntitySizeLimit", 0);
+    saxParser.getXMLReader().setProperty("jdk.xml.totalEntitySizeLimit", 0);
+    saxParser.getXMLReader().setProperty("jdk.xml.entityExpansionLimit", 0);
```

Comment:
> > While the entity expansion limits are good security measures, consider also disabling external DTD loading
>
> @cushon Thanks! Does coderabbit's suggestion make sense?

I'm not sure--I have context about the JDK change, but less about the requirements of this project.

I can try adding the change to disable external DTD loading while we're here, but it's somewhat unrelated. I think coderabbit's is a bit confused--this isn't setting entity expansion limits, it's actually disabling the limits (setting `0` means 'unlimited').

Are the XML files parsed by this code are trusted, either because they're the ones included in this project, or provided by a user of the library, rather than being arbitrary user input?

Reply:
> Are the XML files parsed by this code are trusted

Generally, yes. While there could be external entities, one (= a user who edits the XML files) would need to add them explicitly.

Reply:
Thanks, I went ahead and added the change to disable external DTD loading.